### PR TITLE
fix lastpass import credit card expiration

### PR DESCRIPTION
**Pull request by figadore**
throwing my hat in the ring to fix https://github.com/bitwarden/web/issues/443
builds off #57 

first time using typescript. suggestions welcome

### CODE REVIEW

File: src/importers/lastpassCsvImporter.ts [hunk 1]
```diff
@@ -170,7 +170,28 @@ export class LastPassCsvImporter extends BaseImporter implements Importer {
...
+                    } else {
+                        const [monthString, year] = mappedData[0].expMonth.split(',');
+                        // Parse month name into number
+                        if (monthString) {
+                            const month = new Date(Date.parse(monthString + '1, 2012')).getMonth() + 1;
+                            mappedData[0].expMonth = month.toString();
+                        } else {
+                            delete mappedData[0].expMonth;
+                        }
+                        if (year) {
```
Comment (kspearrin) [inline src/importers/lastpassCsvImporter.ts]: same

File: src/importers/lastpassCsvImporter.ts [hunk 2]
```diff
@@ -170,7 +170,28 @@ export class LastPassCsvImporter extends BaseImporter implements Importer {
                         'Number': 'number',
                         'Name on Card': 'cardholderName',
                         'Security Code': 'code',
+                        // LP provides date in a format like 'June,2020'
+                        // Store in expMonth, then parse and modify
+                        'Expiration Date': 'expMonth',
                     });
+
+                    if (mappedData[0].expMonth === ',') {
```
Comment (kspearrin) [inline src/importers/lastpassCsvImporter.ts]: Is there ever a case where `mappedData[0].expMonth` is just an empty string, or null after `parseSecureNoteMapping`?

Comment (figadore) [reply src/importers/lastpassCsvImporter.ts]: not in my tests. in  the `should parse blank card note` test i added, the data is from a note i exported from lastpass that literally had no fields filled out, and the result had `Expiration Date:,` as a line in the `extra` column

File: src/importers/lastpassCsvImporter.ts (hunk 1 above)
Comment (figadore) [reply src/importers/lastpassCsvImporter.ts]: updated

File: src/importers/lastpassCsvImporter.ts [hunk 3]
```diff
@@ -170,7 +170,28 @@ export class LastPassCsvImporter extends BaseImporter implements Importer {
...
+                        // No expiration data
+                        delete mappedData[0].expMonth;
+                    } else {
+                        const [monthString, year] = mappedData[0].expMonth.split(',');
+                        // Parse month name into number
+                        if (monthString) {
```
[the remark this answers is not in the document]
Comment (figadore) [reply src/importers/lastpassCsvImporter.ts]: updated

File: src/importers/lastpassCsvImporter.ts (hunk 2 above)
Comment (kspearrin) [reply src/importers/lastpassCsvImporter.ts]: Ok, but we still should guard against null reference exceptions here. Lastpass could change something in the future

File: src/importers/lastpassCsvImporter.ts [hunk 4]
```diff
@@ -170,7 +170,29 @@ export class LastPassCsvImporter extends BaseImporter implements Importer {
                         'Number': 'number',
                         'Name on Card': 'cardholderName',
                         'Security Code': 'code',
+                        // LP provides date in a format like 'June,2020'
+                        // Store in expMonth, then parse and modify
+                        'Expiration Date': 'expMonth',
                     });
+
+                    const exp = mappedData[0].expMonth;
```
Comment (figadore) [inline src/importers/lastpassCsvImporter.ts]: not too familiar with the codebase. can we rely on `mappedData[0]` to exist in this context?

File: src/importers/lastpassCsvImporter.ts (hunk 2 above)
Comment (figadore) [reply src/importers/lastpassCsvImporter.ts]: sure thing. updated

File: src/importers/lastpassCsvImporter.ts (hunk 4 above)
Comment (figadore) [reply src/importers/lastpassCsvImporter.ts]: looks like yes

**Pull request merged**